docs/reference/performance.rst

```diff
@@ -0,0 +1,277 @@
+
+Finally, the throughput of memory accesses by a kernel can vary significantly 
+depending on the access pattern for each type of memory. Therefore, the next 
+step in maximizing memory throughput is to organize memory accesses as 
+optimally as possible. This is especially important for global memory accesses,
+as global memory bandwidth is low compared to available on-chip bandwidths and
+arithmetic instruction throughput. Thus, non-optimal global memory accesses
+generally have a high impact on performance.
+
+Data Transfer
+-------------------------------------------------------------------------------
+applications should aim to minimize data transfers between the host and the 
+device. This can be achieved by moving more computations from the host to the 
+device, even if it means running kernels that do not fully utilize the 
+parallelism for device. Intermediate data structures can be created, used, 
+and discarded in device memory without being mapped or copied to host memory.
+
+Batching small transfers into a single large transfer can improve performance 
+due to the overhead associated with each transfer. On systems with a front-side
+bus, using page-locked host memory can enhance data transfer performance.
+
+When using mapped page-locked memory, there is no need to allocate device 
+memory or explicitly copy data between device and host memory. Data transfers 
+occur implicitly each time the kernel accesses the mapped memory. For optimal
+performance, these memory accesses should be coalesced, similar to global
+memory accesses.
+
+On integrated systems where device and host memory are physically the same,
+any copy operation between host and device memory is unnecessary, and mapped
+page-locked memory should be used instead. Applications can check if a device
+is integrated by querying the integrated device property.
+
+
+Device Memory Access
+-------------------------------------------------------------------------------
+Memory access instructions may be repeated due to the spread of memory
+addresses across warp threads. The impact on throughput varies with memory type
+and is generally reduced when addresses are more scattered, especially in
+global memory.
+
+Device memory is accessed via 32-, 64-, or 128-byte transactions that must be
+naturally aligned. Maximizing memory throughput involves coalescing memory
+accesses of threads within a warp into minimal transactions, following optimal
+access patterns, using properly sized and aligned data types, and padding data
+when necessary.
+
+Global memory instructions support reading or writing data of specific sizes 
+(1, 2, 4, 8, or 16 bytes) that are naturally aligned. If the size and alignment
+requirements are not met, it leads to multiple instructions, reducing 
+performance. Therefore, using data types that meet these requirements, ensuring
+alignment for structures, and maintaining alignment for all values or arrays is
+crucial for correct results and optimal performance.
```

Comment:
Hmm, correct results should be obtained even on bad alignment, I suggest
```suggestion
crucial for optimal performance.
```

Reply:
Added